homeassistant/components/deconz/fan.py

```diff
@@ -105,7 +105,11 @@ def async_update_callback(self, force_update=False) -> None:
 
     async def async_set_speed(self, speed: str) -> None:
         """Set the speed of the fan."""
+        if speed not in SPEEDS:
+            return
```

Comment:
Consider logging a warning or raising an error. Raising an error (`ValueError`) will show in the frontend.